File: Data Structures & Algorithms/max-water-container/submission-1.cpp
class Solution {
public:
    int maxArea(vector<int>& heights) {
        int max1 = 0;
        int l = 0;
        int r = heights.size() -1;

        while(l < r){
            int m = 0;
            int a = heights[l];
            int b = heights[r];

            if (a > b) {
                m = (b*(r-l));
                r -= 1;
            }
            else if( b > a) {
                m = (a*(r-l));
                l += 1;

            }

            else{
                m =(a*(r-l));
                r-=1;
            }
            max1 = max( max1 , m );
        }
        return max1;
    }
};

File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def goodNodes(self, root: TreeNode) -> int:
        res = 0

        if not root:
            return res
        

        def dfs(curr , maxval):
            
            nonlocal res
            
            if not curr:
                return


            maxval = max(curr.val , maxval)

            if curr.val >= maxval:
                res += 1

            dfs(curr.left , maxval)
            dfs(curr.right , maxval)        


        dfs(root , root.val)

        return res

            

                